### PR TITLE
Expose Git secret token if default context used

### LINKED ISSUE #135: git token secret should be exposed automatically when building from git
https://github.com/docker/build-push-action#git-context

> If you use this action in a private repository, you have to pass the GitHub Token as a secret named GIT_AUTH_TOKEN to be able to authenticate against it with buildx:

There shouldn't be a need to pass `GIT_AUTH_TOKEN` when building from git, or at least when building from the current repository. I believe GH gives access to this secret always and we should automatically forward it when building from the current repo. Even on public repos I think it is better to use token.

### PR DESCRIPTION
**Pull request by crazy-max**
Closes #135 

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,11 +78,10 @@ jobs:
       -
         name: Image digest
         run: echo ${{ steps.docker_build.outputs.digest }}
-  ```
-</details>
+```
 
-If you use this action in a private repository, you have to pass the [GitHub Token](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token)
-as a secret named `GIT_AUTH_TOKEN` to be able to authenticate against it with buildx:
+If you want to authenticate against a private repository, you have to use a secret named `GIT_AUTH_TOKEN` to be able
```
Comment (tonistiigi) [inline README.md:83]: "against another private repository" ?

Comment (tonistiigi) [reply README.md:83]: "building from current repository automatically uses the token and it does not need to be passed"

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -50,7 +92,7 @@ jobs:
             localhost:5000/name/app:latest
             localhost:5000/name/app:1.0.0
           secrets: |
-            GIT_AUTH_TOKEN=${{ github.token }}
+            GIT_AUTH_TOKEN=01234567890abcdefgh
```
Comment (tonistiigi) [inline .github/workflows/ci.yml:95]: not sure what this is testing or why the above build does not fail with an invalid token

Comment (crazy-max) [reply .github/workflows/ci.yml:95]: I've missed that thanks. I think it passed because it's a public repo anyway.

Comment (tonistiigi) [reply .github/workflows/ci.yml:95]: I think what is missing is that `githubToken` input should be ignored if `GIT_AUTH_TOKEN` is manually defined.

Comment (crazy-max) [reply .github/workflows/ci.yml]: It is through [this cond](https://github.com/docker/build-push-action/pull/136/files#diff-a7af09f7fa8732629ff9579258b5584bR98-R100).

Comment (tonistiigi) [reply .github/workflows/ci.yml]: hmm, maybe this is expected then from the server side. I wonder if you send a wrong auth header to the HTTP requests that does the clone of a public repository it still passes.

Comment (crazy-max) [reply .github/workflows/ci.yml]: I will make a quick test locally

Comment (crazy-max) [reply .github/workflows/ci.yml]: Ok so indeed the server seems to skip the header if the repository is public

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**